### ISSUE TITLE
Background issue

### ISSUE THREAD

**Issue by tspent** [feedfm/iOS-SDK#1]
Hey, 
Every time the app comes to foreground its resumes to play even if we called  -stop manually before we go to background.
The playbackState changes to  FMAudioPlayerPlaybackStatePlaying.

Thanks,

**Comment by eml** [comment]
Hrmm.. I took the sample player here https://github.com/feedfm/iOS-RadioPlayer and added a stop button. When I hit stop and then switch to a different app and back, the player says it is still in FMAudioPlayerPlaybackStateReadyToPlay . I can't reproduce this.

**Comment by averello** [comment]
Hey @eml,

we downloaded the https://github.com/feedfm/iOS-RadioPlayer sample app and we replaced the "Pause button" code with:

```
- (void) onClick {
    if ((_feedPlayer.playbackState == FMAudioPlayerPlaybackStatePaused) ||
        (_feedPlayer.playbackState == FMAudioPlayerPlaybackStateReadyToPlay) ||
        (_feedPlayer.playbackState == FMAudioPlayerPlaybackStateComplete)) {
        [_feedPlayer play];
    } else {
        [_feedPlayer stop]; // <--- here is the change
    }
}
```

and we reproduce the issue reported by @tspent 



Describing the steps to reproduce:

1. Launch iOS-RadioPlayer
2. Press "play"
3. Wait for the music to start playing
4. Press "pause" (that calls -stop, as depicted in the code snippet above)
4. Hit the home button
5. Wait until the sound fades out
6. Tap the App Icon to open the app

Result: the radio starts playing.

Tested on iPhone (iOS 8.3), iPhone 6 (iOS 10.1.1) & iPhone 6 Plus (iOS 10.2.1).
Tested both with and without VPN.

The problem can't be reproduced when `-pause` is called.

**Comment by eml** [comment]
That's what I tried when I was first trying to reproduce this. I tried again, but on a real device instead of the simulator and this time, and I was able to reproduce it. Very odd!

I'll hunt down the bug, but likely won't have a fix until mid to late next week.

**Comment by tspent** [comment]
Hello @eml, 
any update for the issue?

Thanks,

**Comment by eml** [comment]
Hi @tspent - sorry for the long time to follow up. A fix has been implemented and pushed in version 3.0.4 of the CocoaPod.
